### PR TITLE
Fixing Run Functional tasks issue

### PR DESCRIPTION
**Pull request by prawalagarwal**
Problem: Currently default working directory for RunFunctionalTests task on remote test agents is temp directory and the test settigns file is created there. Therefore, if any deployment items are specified within the test settings file with relative path, they are searched within the temp and not in the test drop location.

Issue Link: https://github.com/Microsoft/vsts-tasks/issues/1638

Fix: Checking if the path of depoyment items is not absolute and then prefixing them with test drop location, generating absolute path in the process.

Testing: Manual

### CODE REVIEW

File: Tasks/RunDistributedTests/RunDistributedTests.ps1 [hunk 1]
```diff
@@ -24,6 +24,40 @@ Function CmdletHasMember($memberName) {
...
 }
 
+Function ModifyTestSettingsForDeploymentItems() {
+
+    if([System.String]::IsNullOrWhiteSpace($runSettingsFile) -Or ([string]::Compare([io.path]::GetExtension($runSettingsFile), ".runsettings", $True) -eq 0) -Or (Test-Path $runSettingsFile -pathtype container))
+    {
+        Write-Verbose "No modifications to test settings required"
+    }
+    else{
+        if (([string]::Compare([io.path]::GetExtension($runSettingsFile), ".testsettings", $True) -eq 0)) {
```
Comment (kaadhina) [inline Tasks/RunDistributedTests/RunDistributedTests.ps1]: is this applicable only for testsettings, if so can we move this condition to above

File: Tasks/RunDistributedTests/RunDistributedTests.ps1 [hunk 2]
```diff
@@ -24,6 +24,40 @@ Function CmdletHasMember($memberName) {
...
+            
+            $runSettingsContent = [System.Xml.XmlDocument](Get-Content $runSettingsFile)
+			$runConfigurationElement = $runSettingsContent.SelectNodes("/*/*/*[local-name()='DeploymentItem']")
```
Comment (kaadhina) [inline Tasks/RunDistributedTests/RunDistributedTests.ps1]: can this be null

File: Tasks/RunDistributedTests/RunDistributedTests.ps1 [hunk 3]
```diff
@@ -24,6 +24,40 @@ Function CmdletHasMember($memberName) {
...
+                    if (!([io.path]::IsPathRooted($runConfigurationElement[$index].Attributes["filename"].Value)))
+                    {
+                        $runConfigurationElement[$index].Attributes["filename"].Value = [io.path]::Combine($dropLocation, $runConfigurationElement[$index].Attributes["filename"].Value);                        
+                    }
+                }
+                
+            }
+            
+            $tempFile = [io.path]::GetTempFileName()
+            $runSettingsContent.Save($tempFile)
```
Comment (kaadhina) [inline Tasks/RunDistributedTests/RunDistributedTests.ps1]: can we delete this temp file created after the run

**Comment by allendm-msft** [comment]
kaadhina has given the comments. Address them before pushing. Otherwise changes are fine

File: Tasks/RunDistributedTests/RunDistributedTests.ps1 (hunk 2 above)
Comment (prawalagarwal) [inline Tasks/RunDistributedTests/RunDistributedTests.ps1]: No. The count is 0 but the object is not null

**Pull request merged**